doc/rst/language/spec/records.rst

```diff
@@ -643,30 +643,46 @@ order they are declared in the record definition.
 Hashing a Record
 ~~~~~~~~~~~~~~~~
 
-When a record is the key for a hashtable, including when using it as the index
-type for an associative domain, the compiler will generate a default hash
-function to use. This behavior can be overridden if more control of the
-hashing used is desired. This can be done by defining a ``hash`` method on
-a record.
+For any record that does not have a user-defined ``==`` or ``!=``
+operator, the compiler will automatically define a default hash method
+for it.  This allows values of that record type to be used as the
+indices of an associative domain, the elements of a set, or the keys
+of a map.  The user can override this default hash method (or provide
+one in cases that the compiler does not) by defining their own method
+named ``hash`` on the record which takes no arguments and returns a
+``uint`` or ``int``.
 
-.. code-block:: chapel
+   *Example (userhash.chpl)*.
+
+   .. code-block:: chapel
+
+
+      record R {
+        var i: uint;
+
+        proc hash() {
+          writeln("In custom hash function");
+          return i;
+        }
+      }
+
+      // Creating an associative domain with an 'idxType' of 'R'
+      // invokes R.hash() as part of its implementation
+
+      var D: domain(R);
+      var r = new R(42);
+      D += r;
+      writeln(D);
+
+   .. BLOCK-test-chapeloutput
+
+      In custom hash function
+      {(i = 42)}
 
-   use Map;
-   var m = new map(R, int);
-   proc R.hash() {
-     writeln("In custom hash function");
-     return i;
-   }
-   var myR = new R();
-   // Indexing the map using an instance of R using the R.hash() method
-   m[myR] = 5;
-
-   var myD = domain(R);
-   myD += myR;
-
-Note that the compiler generated ``hash`` can only be overriden on records
-that have been defined in user code, and will not override the compiler hash
-for ``int.hash``, for example.
+Note that the compiler-generated ``hash`` can only be overridden for
+records that have been defined in user code, and cannot be used to
```

Comment:
For grammar - I recommend adding "it" after "and" or else removing the comma. Starting a new sentence would work too.

Reply:
Thanks, I split into two sentences.